### PR TITLE
Dashboard endpoint

### PR DESCRIPTION
**Pull request by laurenfrederick**
**Summary:** Add an endpoint to serve the dashboard from an S3 bucket

Addresses [CUMULUS-1105](https://bugs.earthdata.nasa.gov/browse/CUMULUS-1105)

## Changes

* Added a dashboard/ endpoint to serve the dashboard from an S3 bucket.

## PR Checklist

- [x] Update CHANGELOG
- [x] Unit tests
- [x] Adhoc testing
- [ ] Integration tests

### CODE REVIEW

File: example/iam/config.yml [hunk 1]
```diff
@@ -94,6 +101,12 @@ lf:
     protected-2:
       name: lf-protected-2
       type: protected
+    dashboard:
```
Comment (jennyhliu) [inline example/iam/config.yml]: Are `dashboard` and `dashboard-2` supposed to overwrite the default buckets?  The default bucket types are declared as `sandbox-dashboard` and `sit-dashboard` earlier.

Comment (laurenfrederick) [reply example/iam/config.yml]: yeah I can remove them, I was just using those for my testing.

**Comment by jennyhliu** [comment]
I have the dashboard deployed to the s3 bucket specified in iam config.yml, and updated endpoints are deployed.  How do I run the dashboard?

File: packages/api/endpoints/dashboard.js [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+ * the dashboard from the specified bucket
+ *
+ * @param {Object} req - express request object
+ * @param {Object} res - express response object
+ * @returns {Promise<Object>} the promise of express response object
+ */
+async function get(req, res) {
+  const [bucket, key] = getFileBucketAndKey(req.params[0]);
+
+  const getObjectResponse = await getS3Object(bucket, key);
```
Comment (markdboyd) [inline packages/api/endpoints/dashboard.js]: I don't know if there would be any appreciable performance benefit, but I think you could pipe a stream of the S3 object into the `res` object

File: packages/common/aws.js [hunk 3]
```diff
@@ -1000,3 +1000,34 @@ exports.retryOnThrottlingException = (fn, options) =>
       () => fn(...args).catch(retryIfThrottlingException),
       options
     );
+
+/**
+ * Error class for file locations that are unparsable
+ */
+class UnparsableFileLocationError extends Error {
```
Comment (markdboyd) [inline packages/common/aws.js]: Should this go in `errors.js`?

File: packages/api/endpoints/dashboard.js (hunk 2 above)
Comment (markdboyd) [reply packages/api/endpoints/dashboard.js]: Something like:

`return s3ObjectReadStream.pipe(res)`

Comment (laurenfrederick) [reply packages/api/endpoints/dashboard.js]: But I need to grab the content-type out too, how would I get that?

File: packages/common/aws.js (hunk 3 above)
Comment (laurenfrederick) [reply packages/common/aws.js]: That sounds good, I moved it

File: packages/api/endpoints/dashboard.js (hunk 2 above)
Comment (markdboyd) [reply packages/api/endpoints/dashboard.js]: Apparently you can do something like this:

```
s3.getObject({Bucket: myBucket, Key: myFile})
    .on('httpHeaders', function (code, headers) {
        if (code < 300) {
            res.set(_.pick(headers, 'content-type', 'content-length', 'last-modified'));
        }                            
    })
    .createReadStream()
    .on('error', next)
    .pipe(res);
```

See https://stackoverflow.com/questions/35782434/streaming-file-from-s3-with-express-including-information-on-length-and-filetype

**Pull request merged**